### PR TITLE
fix: update configs types

### PR DESCRIPTION
**Pull request by gurgelio**
Not all config keys were included in the Record, only the recommended ones.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -59,19 +59,13 @@ import validTypes from './rules/validTypes.js';
 import { getJsdocProcessorPlugin } from './getJsdocProcessorPlugin.js';
 
 /**
+ * @typedef {"recommended" | "stylistic" | "contents" | "logical"} ConfigGrous
```
Comment (brettz9) [inline src/index.js:62]:
```suggestion
 * @typedef {"recommended" | "stylistic" | "contents" | "logical"} ConfigGroups
```

File: src/index.js [hunk 2]
```diff
@@ -59,19 +59,13 @@ import validTypes from './rules/validTypes.js';
...
- *       "recommended-typescript-flavor-error"|"flat/recommended"|
- *       "flat/recommended-error"|"flat/recommended-typescript"|
- *       "flat/recommended-typescript-error"|
- *       "flat/recommended-typescript-flavor"|
- *       "flat/recommended-typescript-flavor-error",
- *     import('eslint').Linter.FlatConfig
- *   >
- * }}
- */
+*   configs: Record<`flat/${ConfigGrous}${ConfigVariants}${ErrorLevelVariants}`, import('eslint').Linter.Config>
```
Comment (brettz9) [inline src/index.js:66]:
```suggestion
*   configs: Record<`flat/${ConfigGroups}${ConfigVariants}${ErrorLevelVariants}`, import('eslint').Linter.Config>
```

**Review by brettz9 — CHANGES_REQUESTED** [review]
Thanks... I think just the suggested nit fixes and we should be good to go here...

File: src/index.js (hunk 2 above)
Comment (brettz9) [reply src/index.js:66]: This change apparently wasn't committed.

**Comment by gurgelio** [comment]
I've just fixed those typos

**Review by brettz9 — APPROVED** [review]

**Comment by brettz9** [comment]
Thanks for the PR!

**Pull request merged**